### DISCUSSION TITLE
start browsers minimized?

### DISCUSSION THREAD

**Discussion by FlyingToto** [jef/streetmerchant#1415] [General]
just an idea.... 
it would be interesting to see if, while runing headless=false, you could have the script start the browser and keep the tab minimized....

**Comment by absolutelyNoClue** [comment]
That would be really helpful! One way to keep it minimized seems to be by setting the defaultViewPort to null and/or the width and height to a very small value as described in the puppeteer launch options, but I'm not able to make it run. Did anyone try this already?

**Comment by FlyingToto** [comment]
while checking chromium docs, I noticed a few command line options which might do the trick:
reference: [https://peter.sh/experiments/chromium-command-line-switches/](url)


--ash-host-window-bounds ⊗ | Sets a window size, optional position, and optional scale factor. "1024x768" creates a window of size 1024x768. "100+200-1024x768" positions the window at 100,200. "1024x768*2" sets the scale factor to 2 for a high DPI display. "800,0+800-800x800" for two displays at 800x800 resolution. "800,0+800-800x800,0+1600-800x800" for three displays at 800x800 resolution. ↪



--window-position ⊗ | Specify the initial window position: --window-position=x,y ↪
-- | --
--window-size ⊗ | Sets the initial window size. Provided as string in the format "800,600". ↪

**Comment by FlyingToto** [comment]
in fact if indeed the switches work, maybe you can just let people pick whatever value in the config file... for instance in my case, I just move the window to the bottom of my screen and just let the tab titles stick out while the rest remain hidden...
